probe-rs/src/debug/unit_info.rs

```diff
@@ -108,9 +108,9 @@ impl UnitInfo {
 
     /// Check if the function located at the given offset contains inlined functions at the
     /// given address.
-    pub(crate) fn find_inlined_functions(
-        &self,
-        debug_info: &DebugInfo,
+    pub(crate) fn find_inlined_functions<'debug_info>(
+        &'debug_info self,
```

Comment:
@noppej would you please try and add this lifetime to wherever it's needed in the output? It's a bit confusing at first read (looks like it should be elided instead, which just doesn't compile)

Reply:
I know what you mean by it seems like it should be elided, but not sure I understand what you mean by "please try and add this to wherever it's needed" ... can you explain please?

Reply:
If I see correctly, `FunctionDie` has two lifetime args, one or both of which should be spelled out as `'debug_info`, no?

Reply:
I'm afraid I still don't understand. My approach is to NOT be explicit about lifetimes when the compiler/rust-analyzer can figure it out. Am I missing something important in what you're seeing?

Reply:
My point is that these two lifetimes are spelled out because something in the return type needs them to be. However, it's not at all obvious what needs them and why they can't be elided. I'm looking at this piece of code and if I don't look up how FunctionDie looks, I can't tell why clippy isn't throwing warnings around. Not a big deal, just a piece of surprise for the reader who thinks they can simplify this :)

Reply:
Gotcha. If I understand your point, you actually have to look all the way into `gimli::DebuggingInformationEntry` to see the required lifetimes. I've worked with that library enough to have a bit of a sense of how it works, but in the beginning it drove me dizzy. On the positive side, that library really is lightning fast, so I can see the value of all the lifetimes.

Reply:
@bugadani Here is an alternate version that will also compile, but feels considerably clunkier to me personally. What are your thoughts?
```rust
    pub(crate) fn find_inlined_functions<'abbrev, 'unit>(
        &'abbrev self,
        debug_info: &'abbrev DebugInfo,
        address: u64,
        offset: UnitOffset,
    ) -> Result<Vec<FunctionDie<'abbrev, 'unit>>, DebugError>
    where
        'abbrev: 'unit,

```

Reply:
While definitely more verbose, I appreciate the unambiguiuty of this version.

This should also work and is enough detail for readability:

```rust
    pub(crate) fn find_inlined_functions<'abbrev>(
        &'abbrev self,
        debug_info: &'abbrev DebugInfo,
        address: u64,
        offset: UnitOffset,
    ) -> Result<Vec<FunctionDie<'abbrev, '_>>, DebugError> {
```

Reply:
A good compromise/improvement. Moving this to draft until I fix it.